pkg/controller/installation/core_controller.go

```diff
@@ -1843,11 +1851,20 @@ func isOpenshiftOnAws(install *operator.Installation, ctx context.Context, clien
 	infra := configv1.Infrastructure{}
 	// If configured to run in openshift, then also fetch the openshift configuration API.
 	if err := client.Get(ctx, types.NamespacedName{Name: openshiftNetworkConfig}, &infra); err != nil {
-		return false, fmt.Errorf("unable to read OpenShift infrastructure configuration: %s", err.Error())
+		return false, fmt.Errorf("unable to read OpenShift infrastructure configuration: %w", err)
 	}
 	return (infra.Status.PlatformStatus.Type == "AWS"), nil
 }
 
+// isHostedOpenShift returns true if this cluster is an OpenShift HCP hosted cluster.
+func isHostedOpenShift(ctx context.Context, client client.Client) (bool, error) {
+	infra := configv1.Infrastructure{}
+	if err := client.Get(ctx, types.NamespacedName{Name: openshiftNetworkConfig}, &infra); err != nil {
```

Comment:
If you don't start a Watch() for this resource, then there will be nothing in the client's cache and we'll need to go round-trip to the API anyway.

You should add a Watch in this block: https://github.com/tigera/operator/blob/01ec8648080b1da2388df5997564e0a16c9b5646/pkg/controller/installation/core_controller.go#L216-L225